timemachine/cpp/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 project(timemachine LANGUAGES CXX CUDA)
 
-find_package(Python 3.10...3.12.8 COMPONENTS Interpreter Development REQUIRED)
+find_package(Python 3.10...<3.13 COMPONENTS Interpreter Development REQUIRED)
```

Comment:
Should this be consistent with https://github.com/proteneer/timemachine/blob/94ee2190911803c42954e0c1277854d129a265b7/setup.py#L94?

I'd be in favor of requiring >=3.12 consistently with what's currently in `setup.py` so that we can adopt some of the newer syntax features.